File: backend/src/main/java/com/streaker/controller/habit/dto/HabitResponseDto.java
package com.streaker.controller.habit.dto;


import com.streaker.utlis.enums.Frequency;

import java.time.LocalDateTime;
import java.util.UUID;

public record HabitResponseDto(
        UUID uuid,
        String name,
        String description,
        Frequency frequency,
        LocalDateTime createdAt,
        UUID userId,
        UUID categoryId,
        UUID streakId
) {}

File: backend/src/main/java/com/streaker/controller/habit/dto/HabitResponseDto.kt
package com.streaker.controller.habit.dto

import com.streaker.utlis.enums.Frequency
import java.time.LocalDateTime
import java.util.UUID

class HabitResponseDto(
    val uuid: UUID,
    val name: String,
    val description: String,
    val frequency: Frequency,
    val createdAt: LocalDateTime,
    val userId: UUID,
    val categoryId: UUID,
    val streakId: UUID
) {}